docs/blog/2025-01-29-new-nx-experience.md

```diff
@@ -0,0 +1,343 @@
+
+The core difference to understand here is that, traditionally, you might have a single task that combines building and type checking, or at least two separate ones (like for Vite applications): one for building and one for type checking. For large TypeScript monorepos, this approach can become problematic because type checking occurs at the application level for all projects in the repository. This means a single TypeScript program is responsible for the entire process.
+
+{% video-player src="/documentation/blog/media/animation-tsrefs-splitting.mp4" alt="Moving from a single build + typecheck to fine-grained distributable typechecks" showDescription=true autoPlay=true loop=true /%}
+
+This has two major implications:
+
+- It cannot be parallelized and distributed on CI, which makes it slow.
+- It is memory-intensive, requiring large CI machines (resulting in higher costs).
+
+By leveraging the new Nx and TypeScript project references setup, we now have fine-grained, individual type checking tasks at the project level. This allows for much better distribution as these tasks can be spread across different [Nx Agents](/ci/features/distribute-task-execution).
+
+![Running type checking on different agents on CI](/blog/images/articles/cipe-typecheck-distribution.avif)
+
+As such, the new approach:
+
+- Is **distributable across multiple machines**, resulting in faster CI runs.
+- Each individual type check **requires less memory**, allowing the use of smaller machines.
+- Each **individual type check can be [cached](/features/cache-task-results)**, significantly speeding up follow-up runs.
+
+We **ran some benchmarks** comparing [the current Nx setup](https://github.com/jaysoo/ts-bench-old) and [the new TypeScript project references-based setup](https://github.com/jaysoo/ts-bench-new):
+
+- Current: [~11m 54s](https://github.com/jaysoo/ts-bench-old/actions/runs/12956714862)
+- New TypeScript project references setup: [~8m 8s](https://github.com/jaysoo/ts-bench-new/actions/runs/12956715557)
+
+And this is the **worst-case scenario**, relying solely on the distribution of type checking tasks across different machines. Subsequent runs might hit the cache and become even faster.
+
+It is important to emphasize that the main reason we can fully leverage these benefits from TypeScript project references is that we avoid the maintenance burden of setting them up manually, thanks to the automated [Nx sync](#automatically-syncing-typescript-project-references) command.
+
+**Struggling with performance in your large TypeScript monorepo?** [Let us know](https://nx.dev/contact/engineering). We’ve [worked with many teams](/customers) to solve similar challenges and would be happy to help. [Reach out!](/contact/engineering)
```

Comment:
Is there any reason one link is `https://nx.dev/contact/engineering` and the other is `/contact/engineering`? Is